django/core/management/commands/shell.py

```diff
@@ -111,6 +127,55 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_app_name(self, app_module):
+        rparts = list(reversed(app_module.split(".")))
+        try:
+            try:
+                return rparts[rparts.index(MODELS_MODULE_NAME) + 1]
+            except ValueError:
+                return rparts[1]
+        except IndexError:
+            return app_module
+
+    def get_apps_and_models(self):
+        for app_conf in apps.get_app_configs():
+            if app_conf.models_module:
+                yield app_conf.models_module, app_conf.get_models()
+
+    def get_namespace(self, options, style):
+        """Override this method with import_items in order to import extra_things"""
+
+        imported_objects = {}
+
+        for app_module, app_models in self.get_apps_and_models():
+            if not app_models:
+                continue
+            model_labels = []
+            for model in app_models:
+                if model.__module__:
+                    try:
+                        imported_objects[model.__name__] = import_string(
+                            "%s.%s" % (model.__module__, model.__name__)
+                        )
+                        model_labels.append(model.__name__)
+                    except ImportError as e:
+                        if options.get("traceback"):
+                            traceback.print_exc()
+                        else:
+                            print(
+                                style.ERROR(
+                                    "Failed to import %s from %s, reason %s"
+                                    % (model.__name__, model.__module__, e)
+                                )
+                            )
+            if options.get("display"):
+                self.stdout.write(
+                    f'from {app_module.__name__} import {", ".join(model_labels)}',
+                    self.style.SUCCESS,
+                )
```

Comment:
As I commented on the forum, I don’t think we should add this verbose display, even optionally. Instead, we can have a simple “X objects automatically imported.” message. Users can use `globals()` or shell-specific features to list all imports, which we can perhaps document.